torchgeo/models/croma.py

```diff
@@ -0,0 +1,644 @@
+                nn.ModuleList(
+                    [
+                        Attention(dim=dim, num_heads=num_heads, dropout=attn_dropout),
+                        CrossAttention(
+                            dim=dim, num_heads=num_heads, dropout=attn_dropout
+                        ),
+                        FFN(dim=dim, mult=ff_mult, dropout=ff_dropout),
+                    ]
+                )
+            )
+
+        self.norm_out = nn.LayerNorm(dim)
+
+    def forward(
+        self, x: Tensor, context: Tensor, relative_position_bias: Tensor
+    ) -> Tensor:
+        """Forward pass of the base transformer model with cross-attention.
+
+        Args:
+            x: Input tensor.
+            context: Context tensor.
+            relative_position_bias: Relative position bias tensor.
+
+        Returns:
+            Output tensor.
+        """
+        for self_attn, cross_attn, ffn in self.layers:
+            x = self_attn(x, relative_position_bias) + x
+            x = cross_attn(x, context, relative_position_bias) + x
+            x = ffn(x) + x
+
+        x = self.norm_out(x)
+        return x
+
+
+class ViT(nn.Module):
+    """Vision Transformer model."""
+
+    def __init__(self, dim: int, depth: int, in_channels: int) -> None:
+        """Initialize the vision transformer model.
+
+        Args:
+            dim: Dimension of the input.
+            depth: Depth of the transformer.
+            in_channels: Number of input channels.
+        """
+        super().__init__()
+        self.depth = depth
+        self.in_channels = in_channels
+        self.dim = dim
+        self.num_heads = 16  # always 16, for base and large models
+        self.patch_size = 8  # always 8, for base and large models
+
+        pixels_per_patch = int(self.patch_size * self.patch_size * in_channels)
+        self.linear_input = nn.Linear(pixels_per_patch, self.dim)
+        self.transformer = BaseTransformer(
+            dim=self.dim, depth=self.depth, num_heads=self.num_heads
+        )
+
+    def forward(self, imgs: Tensor, attn_bias: Tensor) -> Tensor:
+        """Forward pass of the vision transformer model.
+
+        Args:
+            imgs: Input tensor.
+            attn_bias: Relative position bias tensor.
+
+        Returns:
+            Output tensor.
+        """
+        x = rearrange(
+            imgs,
+            'b c (h i) (w j) -> b (h w) (c i j)',
+            i=self.patch_size,
+            j=self.patch_size,
+        )
+        # x is shape -> (bsz, num_patches, self.channels*self.patch_size*self.patch_size)
+
+        x = self.linear_input(x)
+        x = self.transformer(x, relative_position_bias=attn_bias)
+        return x
+
+
+class CROMABase_Weights(WeightsEnum):  # type: ignore[misc]
+    """CROMA base model weights.
+
+    .. versionadded:: 0.7
+    """
+
+    CROMA_VIT = Weights(
+        url='https://hf.co/torchgeo/croma/resolve/17591bcf0711940cc71167d6618d74f60847b43b/CROMA_base- 0238d814.pt',
```

Comment:
```suggestion
        url='https://hf.co/torchgeo/croma/resolve/17591bcf0711940cc71167d6618d74f60847b43b/CROMA_base-0238d814.pt',
```

Reply:
Oh wait, the file you uploaded also contains this space. Can you rename the uploaded file and update the commit hashes in the URL?